tools/pythonpkg/src/pyrelation.cpp

```diff
@@ -401,6 +409,100 @@ unique_ptr<DuckDBPyRelation> DuckDBPyRelation::CumMin(const string &aggr_columns
 	return GenericWindowFunction("min", aggr_columns);
 }
 
+unique_ptr<DuckDBPyRelation> DuckDBPyRelation::RowNumber(const string &window_spec, const string &projected_columns) {
+	auto expr = GenerateExpressionList("row_number", "*", "", "", false, projected_columns, window_spec);
+	return make_uniq<DuckDBPyRelation>(rel->Project(expr));
```

Comment:
Slight nitpick but is there a reason we don't use GenericWindowFunction in these functions?
Currently these methods are only 2 lines long but if this scales in the future we would be duplicating these changes across all these functions, so I'd prefer to turn these into just a call to GenericWindowFunction if possible

Reply:
Fair point. I will try to refactor this.